Composer/packages/lib/bot-deploy/src/botProjectDeploy.ts

```diff
@@ -300,170 +357,186 @@ export class BotProjectDeploy {
+          })
+        );
 
-      for (let luisConfigFile of luisConfigFiles) {
-        const luisSettings = await fs.readJson(luisConfigFile);
-        Object.assign(luisAppIds, luisSettings.luis);
-      }
+        const loadResult = await builder.loadContents(
+          modelFiles,
+          language || '',
+          environment || '',
+          luisAuthoringRegion || ''
+        );
 
-      const luisEndpoint = `https://${luisAuthoringRegion}.api.cognitive.microsoft.com`;
-      let luisConfig: any = {
-        endpoint: luisEndpoint,
-        endpointKey: luisEndpointKey,
-      };
+        const buildResult = await builder.build(
+          loadResult.luContents,
+          loadResult.recognizers,
+          luisAuthoringKey,
+          luisAuthoringRegion,
+          name,
+          environment,
+          language,
+          false,
+          loadResult.multiRecognizers,
+          loadResult.settings
+        );
+        await builder.writeDialogAssets(buildResult, true, this.generatedFolder);
+
+        this.logger({
+          status: BotProjectDeployLoggerType.DEPLOY_INFO,
+          message: `lubuild succeed`,
+        });
+
+        const luisConfigFiles = (await this.getFiles(this.remoteBotPath)).filter(filename =>
+          filename.includes('luis.settings')
+        );
+        const luisAppIds: any = {};
 
-      Object.assign(luisConfig, luisAppIds);
+        for (const luisConfigFile of luisConfigFiles) {
+          const luisSettings = await fs.readJson(luisConfigFile);
+          Object.assign(luisAppIds, luisSettings.luis);
+        }
 
-      let settings: any = await fs.readJson(this.deploymentSettingsPath);
-      settings['luis'] = luisConfig;
+        const luisEndpoint = `https://${luisAuthoringRegion}.api.cognitive.microsoft.com`;
+        const luisConfig: any = {
+          endpoint: luisEndpoint,
+          endpointKey: luisEndpointKey,
+        };
 
-      await fs.writeJson(this.deploymentSettingsPath, settings);
-      const token = await this.creds.getToken();
+        Object.assign(luisConfig, luisAppIds);
 
-      const getAccountUri = `${luisEndpoint}/luis/api/v2.0/azureaccounts`;
-      const options = {
-        headers: { Authorization: `Bearer ${token.accessToken}`, 'Ocp-Apim-Subscription-Key': luisAuthoringKey },
-      } as rp.RequestPromiseOptions;
-      const response = await rp.get(getAccountUri, options);
-      const jsonRes = JSON.parse(response);
-      const account = this.getAccount(jsonRes, `${name}-${environment}-luis`);
+        // Update deploymentSettings with the luis config
+        const settings: any = await fs.readJson(this.deploymentSettingsPath);
+        settings.luis = luisConfig;
 
-      for (let k in luisAppIds) {
-        const luisAppId = luisAppIds[k];
-        this.logger({
-          status: BotProjectDeployLoggerType.DEPLOY_INFO,
-          message: `Assigning to luis app id: ${luisAppIds}`,
+        await fs.writeJson(this.deploymentSettingsPath, settings, {
```

Comment:
why only write to the build path?

Shouldn't these new LUIS configuration items be written to the main appsettings.json file as well?

Reply:
done

Reply:
I return the result in this script and save in provision script.

Reply:
outdated.